openpype/hosts/webpublisher/plugins/publish/collect_published_files.py

```diff
@@ -0,0 +1,266 @@
+
+            self.log.info("instance.data:: {}".format(instance.data))
+
+    def _get_subset_name(self, family, subset_template, task_name, variant):
+        fill_pairs = {
+            "variant": variant,
+            "family": family,
+            "task": task_name
+        }
+        subset = subset_template.format(**prepare_template_data(fill_pairs))
+        return subset
+
+    def _get_single_repre(self, task_dir, files, tags):
+        _, ext = os.path.splitext(files[0])
+        repre_data = {
+            "name": ext[1:],
+            "ext": ext[1:],
+            "files": files[0],
+            "stagingDir": task_dir,
+            "tags": tags
+        }
+        self.log.info("single file repre_data.data:: {}".format(repre_data))
+        return [repre_data]
+
+    def _process_sequence(self, files, task_dir, tags):
+        """Prepare reprentations for sequence of files."""
+        collections, remainder = clique.assemble(files)
+        assert len(collections) == 1, \
+            "Too many collections in {}".format(files)
+
+        frame_start = list(collections[0].indexes)[0]
+        frame_end = list(collections[0].indexes)[-1]
+        ext = collections[0].tail
+        repre_data = {
+            "frameStart": frame_start,
+            "frameEnd": frame_end,
+            "name": ext[1:],
+            "ext": ext[1:],
+            "files": files,
+            "stagingDir": task_dir,
+            "tags": tags
+        }
+        self.log.info("sequences repre_data.data:: {}".format(repre_data))
+        return [repre_data]
+
+    def _get_family(self, settings, task_type, is_sequence, extension):
+        """Guess family based on input data.
+
+            Args:
+                settings (dict): configuration per task_type
+                task_type (str): Animation|Art etc
+                is_sequence (bool): single file or sequence
+                extension (str): without '.'
+
+            Returns:
+                (family, [families], subset_template_name, tags) tuple
+                AssertionError if not matching family found
+        """
+        task_obj = settings.get(task_type)
+        assert task_obj, "No family configuration for '{}'".format(task_type)
+
+        found_family = None
+        for family, content in task_obj.items():
+            if is_sequence != content["is_sequence"]:
+                continue
+            if extension in content["extensions"] or \
+                    '' in content["extensions"]:  # all extensions setting
+                found_family = family
+                break
+
+        msg = "No family found for combination of " +\
+              "task_type: {}, is_sequence:{}, extension: {}".format(
+                  task_type, is_sequence, extension)
+        assert found_family, msg
+
+        return found_family, \
+            content["families"], \
+            content["subset_template_name"], \
+            content["tags"]
+
+    def _get_version(self, asset_name, subset_name):
```

Comment:
I would rename it to `_get_last_version` as I would expect that this returns version that will be used.